python/ray/train/data_parallel_trainer.py

```diff
@@ -451,13 +451,21 @@ def training_loop(self) -> None:
             checkpoint_config=self.run_config.checkpoint_config,
         )
 
+        def clear_lazy_checkpoint_marker():
+            marker_file = Path(trial_info.logdir) / LAZY_CHECKPOINT_MARKER_FILE
+            if marker_file.exists():
+                logger.debug(
+                    f"Deleting the stale lazy checkpoint marker file: {marker_file}."
+                )
+                marker_file.unlink()
+
+        # Start the remote actors.
+        backend_executor.start(initialization_hook=clear_lazy_checkpoint_marker)
```

Comment:
Can you comment that this needs to happen before the checkpoint manager gets created? 🤯

Reply:
Also, interesting that we had this initialization hook!

Reply:
Because it's TuneCheckpoint Manager that drops the marker：）

(Yeah, surprised we left a back door for our future selves.

Reply:
Oh I meant, could you actually add this as a comment in the code? That the ordering here matters? Or honestly for clarity we might want to move the marker creation here as well.

Reply:
Done!